### ISSUE TITLE
Invalid Opcode

### ISSUE THREAD

**Issue by gosbank** [syuilo/Misskey#8238]
## 💡 Summary

When running misskey service, the node fails with invalid opcode error.

Jan 31 15:23:49 s189866 misskey[4021]: ERR  *#011[core cluster]#011[215] died :(
Jan 31 15:23:53 s189866 kernel: [72827.784142] traps: Misskey (worker[6983] trap invalid opcode ip:7efcea321dfb sp:7ffe95962400 error:0 in libvips-cpp.so.42[7efcea014000+aac000]

## 🙂 Expected Behavior

It should run without error.

## ☹️ Actual Behavior

It crashes due to invalid opcode.

## 📝 Steps to Reproduce

1. Build misskey (https://github.com/miaocloud/misskey/blob/master/docs/setup.en.md)
2. Start from service
3. Look at /var/log/syslog

## 📌 Environment

Ubuntu 18.04, Node 16.13.2, MongoDB shell version v3.6.3
Linux s189866 4.15.0-106-generic #107-Ubuntu SMP Thu Jun 4 11:27:52 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**Comment by Johann150** [comment]
Sorry, but you landed in the wrong repository as I can see from the link you posted in the 1st step. That repository still has Misskey major version 10 and is quite outdated.

**Comment by gosbank** [comment]
This is literally the newest version. I git pulled it an hour ago.

**Comment by Johann150** [comment]
If you pulled it from the repository you posted, you pulled it from a different repository that has an old version in it. Which repository did you pull from?

**Comment by robflop** [comment]
The repository you linked and presumably pulled from is (very) outdated, is what Johann is saying.
If you pull from this repository, you will get the actual newest version.

**Comment by gosbank** [comment]
Great! I pulled from here, but I followed the wrong install instructions. Let me try again.

**Comment by Johann150** [comment]
You can find current installation instructions at https://misskey-hub.net/en/docs/install.html

**Comment by gosbank** [comment]
Those build instructions result in the same problem.

**Comment by Johann150** [comment]
If you did follow those instructions can you please update your initial post with the Postgres & Misskey version you are using?

**Comment by gosbank** [comment]
Postgres is v10.0. 

Because I was looking at the wrong install instructions initially I didn't notice that the new release requires Postgres 12 or 13. Please close this as solved. Or however you want to do it. Apologies for the confusion on my part.

**Comment by gosbank** [comment]
Although, it is also happening on older versions that are compatible with my system. eg. 12.79.1

**Comment by gosbank** [comment]
Jan 31 16:27:12 s189866 misskey[902]: > misskey@12.79.1 start /home/misskey/misskey
Jan 31 16:27:12 s189866 misskey[902]: > node ./index.js
Jan 31 16:27:14 s189866 misskey[902]:   _____ _         _
Jan 31 16:27:14 s189866 misskey[902]:  |     |_|___ ___| |_ ___ _ _
Jan 31 16:27:14 s189866 misskey[902]:  | | | | |_ -|_ -| '_| -_| | |
Jan 31 16:27:14 s189866 misskey[902]:  |_|_|_|_|___|___|_,_|___|_  |
Jan 31 16:27:14 s189866 misskey[902]:  v12.79.1                |___|
Jan 31 16:27:14 s189866 misskey[902]:  Misskey is an open-source decentralized microblogging platform.
Jan 31 16:27:14 s189866 misskey[902]:  If you like Misskey, please donate to support development. https://www.patreon.com/syuilo
Jan 31 16:27:14 s189866 misskey[902]: --- s189866 (PID: 913) ---
Jan 31 16:27:14 s189866 misskey[902]: INFO *#011[core boot]#011Welcome to Misskey!
Jan 31 16:27:14 s189866 misskey[902]: INFO *#011[core boot]#011Misskey v12.79.1
Jan 31 16:27:14 s189866 misskey[902]: INFO *#011[core boot env]#011NODE_ENV: production
Jan 31 16:27:14 s189866 misskey[902]: INFO *#011[core boot env]#011You do not have root privileges
Jan 31 16:27:14 s189866 misskey[902]: INFO *#011[core boot nodejs]#011Version 16.13.2
Jan 31 16:27:14 s189866 misskey[902]: DONE *#011[core boot config]#011Loaded
Jan 31 16:27:14 s189866 misskey[902]: INFO *#011[core boot db]#011Connecting...
Jan 31 16:27:14 s189866 misskey[902]: DONE *#011[core boot db]#011Connected: v10.19 (Ubuntu 10.19-0ubuntu0.18.04.1)
Jan 31 16:27:15 s189866 misskey[902]: DONE *#011[core boot]#011Misskey initialized
Jan 31 16:27:15 s189866 misskey[902]: INFO *#011[core boot]#011Starting 1 worker...
Jan 31 16:27:23 s189866 kernel: [76637.055361] traps: Misskey (worker[925] trap invalid opcode ip:7fb590974f1f sp:7ffdd5fbaf68 error:0 in libvips-cpp.so.42[7fb590742000+c9e000]
Jan 31 16:27:23 s189866 misskey[902]: ERR  *#011[core cluster]#011[1] died :(
Jan 31 16:27:27 s189866 kernel: [76641.695339] traps: Misskey (worker[952] trap invalid opcode ip:7f4580f72f1f sp:7ffe5f7332a8 error:0 in libvips-cpp.so.42[7f4580d40000+c9e000]
Jan 31 16:27:27 s189866 misskey[902]: ERR  *#011[core cluster]#011[2] died :(
Jan 31 16:27:32 s189866 kernel: [76646.308292] traps: Misskey (worker[977] trap invalid opcode ip:7fda41976f1f sp:7ffdd5725c78 error:0 in libvips-cpp.so.42[7fda41744000+c9e000]
Jan 31 16:27:32 s189866 misskey[902]: ERR  *#011[core cluster]#011[3] died :(

**Comment by tamaina** [comment]
Have you installed build-essential?

**Comment by gosbank** [comment]
Yes, I believe all requirements are met. Today I will try even older versions until I find something that builds.

**Comment by gosbank** [comment]
12.10.1 seems to be ok. I'm hesitant to do any upgrades from this point on. So I supposed we're done here.

**Comment by Johann150** [comment]
In any case I think since Misskey does not run on the operating system itself but is run by Node.js, this is probably not something Misskey itself can solve.
